src/status_im2/common/standard_auth/keycard/view.cljs

```diff
@@ -0,0 +1,165 @@
+(ns status-im2.common.standard-auth.keycard.view
+  (:require [react-native.core :as rn]
+            [quo2.core :as quo]
+            [status-im2.common.standard-auth.keycard.style :as style]
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [status-im2.common.resources :as resources]
+            [utils.i18n :as i18n]
+            [quo2.components.keycard.view :as keycard]
+            [utils.re-frame :as rf]))
+
+(defn handle-number-entry
+  [entered-numbers max-digits max-attempt-reached number]
+  (when (>= (count @entered-numbers) max-digits)
+    (reset! entered-numbers []))
+  (when (= (count @entered-numbers) (dec max-digits))
+    (reset! max-attempt-reached true))
+  (swap! entered-numbers conj number))
+
+(defn enter-keycard-pin-sheet
+  [& {:keys [max-digits] :or {max-digits 6}}]
```

Comment:
I think we don't need to complicate the function signature by making it variadic. It just needs a single map of props.